docs/examples.md

```diff
@@ -164,9 +164,146 @@ object Main extends CommandIOApp("mkString", "Concatenates strings from stdin")
+  given csvRowDecoder: CsvRowDecoder[Passenger, String] with
+    def apply(row: CsvRow[String]): DecoderResult[Passenger] =
+      for
+        id <- row.as[Long]("id")
+        firstName <- row.as[String]("First Name")
+        ageOpt <- row.asNonEmpty[Int]("Age")
+        flightNumber <- row.as[String]("flight number")
+        destination <- row.as[String]("destination")
+      yield
+        val age = ageOpt.toRight[String]("N/A")
+        Passenger(id, firstName, age, flightNumber, destination)
+
+    // Here we define a manual encoder for encoding Passenger classes to a CSV
+    given csvRowEncoder: CsvRowEncoder[Passenger, String] with
+      def apply(p: Passenger): CsvRow[String] =
+        CsvRow.fromNelHeaders(
+          NonEmptyList.of(
+            (p.firstName, "first_name"),
+            (p.age.toString(), "age"),
+            (p.flightNumber, "flight_number"),
+            (p.destination, "destination")
+          )
+        )
+
+val input = Files[IO]
+  .readAll(Path("./example.csv"), 1024, Flags.Read)
+  .through(text.utf8.decode)
+  .through(decodeUsingHeaders[Passenger]())
+
+object CSVPrinter extends IOApp.Simple:
+  val run: IO[Unit] =
+    input.evalTap(p => IO.println(p)).compile.drain
+```
+
+
+@:choice(scala-2)
+```scala mdoc:reset:silent
+//> using lib "org.typelevel::toolkit::@VERSION@"
+
+import cats.effect.*
+import fs2.text
+import fs2.data.csv.*
+import fs2.data.csv.generic.semiauto._
+import fs2.io.file.{Path, Flags, Files}
+import cats.data.NonEmptyList
+
+case class Passenger(
+    id: Long,
+    firstName: String,
+    age: Either[String, Int],
+    flightNumber: String,
+    destination: String
+)
+
+object Passenger {
+  // Here we define a manual decoder for each row in our CSV
+  implicit val csvRowDecoder: CsvRowDecoder[Passenger, String] =
+    new CsvRowDecoder[Passenger, String] {
+      def apply(row: CsvRow[String]): DecoderResult[Passenger] =
+        for
+          id <- row.as[Long]("id")
+          firstName <- row.as[String]("First Name")
+          ageOpt <- row.asNonEmpty[Int]("Age")
+          flightNumber <- row.as[String]("flight number")
+          destination <- row.as[String]("destination")
+        yield
+          val age = ageOpt.toRight[String]("N/A")
+          Passenger(id, firstName, age, flightNumber, destination)
+    }
+
+  // Here we define a manual encoder for encoding Passenger classes to a CSV
+  implicit val csvRowEncoder: CsvRowEncoder[Passenger, String] =
+    new CsvRowEncoder[Passenger, String] {
+      def apply(p: Passenger): CsvRow[String] =
+        CsvRow.fromNelHeaders(
+          NonEmptyList.of(
+            (p.firstName, "first_name"),
+            (p.age.toString(), "age"),
+            (p.flightNumber, "flight_number"),
+            (p.destination, "destination")
+          )
+        )
+    }
+}
+
+val input = Files[IO]
+  .readAll(Path("./example.csv"), 1024, Flags.Read)
+  .through(text.utf8.decode)
+  .through(decodeUsingHeaders[Passenger]())
```

Comment:
Are you outside an object here, right? I'm not sure this actually works in scala 2, mdoc compiled it since it's using scala 3. It's maybe worth trying to compile the whole script locally adding a `//> using scala "2.13.10"` directive